Objective-C/CBLDocument.mm

```diff
@@ -109,107 +225,53 @@ - (BOOL) deleteDocument: (NSError**)outError {
 
 
 - (BOOL) purge: (NSError**)outError {
-    if (!self.exists)
-        return NO;
+    assert(_database && _c4db);
+    
+    if (!self.exists) {
+        return createError(kCBLStatusNotFound, outError);
+    }
     
     C4Transaction transaction(_c4db);
     if (!transaction.begin())
         return convertError(transaction.error(),  outError);
     
     C4Error err;
-    if (c4doc_purgeRevision(_c4doc, C4Slice(), &err) >= 0) {
-        if (c4doc_save(_c4doc, 0, &err)) {
+    if (c4doc_purgeRevision(self.c4Doc.rawDoc, C4Slice(), &err) >= 0) {
+        if (c4doc_save(self.c4Doc.rawDoc, 0, &err)) {
             // Save succeeded; now commit:
-            if (!transaction.commit()) {
+            if (!transaction.commit())
                 return convertError(transaction.error(), outError);
-            }
-            
-            // Reload:
-            if (![self loadDoc_mustExist: NO error: outError])
-                return NO;
-            
-            self.properties = nil;
-            [self resetChangesKeys];
             
+            // Reset:
+            [self setC4Doc: nil];
```

Comment:
When calling setC4Doc: method, the raw C4Document object will be wrapped by the CBLC4Document which will free C4Document object when it gets dealloc.